src/AbstractSQLRules2SQL.ts

```diff
@@ -494,7 +509,7 @@ const SubtractDateDuration: MatchFn = (args, indent) => {
 const SubtractDateNumber: MatchFn = (args, indent) => {
 	checkArgs('SubtractDateNumber', args, 2);
 	const a = DateValue(getAbstractSqlQuery(args, 0), indent);
-	const b = NumericValue(getAbstractSqlQuery(args, 1), indent);
+	const b = mathOpValue(args, 1, indent);
```

Comment:
same applies here,  with the special case that the SubtractDateDate returns an integer on postgres